string fromNode
string toNode
float32 battery
int32 idx